AuthenticationAndAuthorization/BusinessLogic/Services/JwtService.cs

```diff
@@ -0,0 +1,51 @@
+﻿using DataAccounts;
+using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.Tokens;
+using System.IdentityModel.Tokens.Jwt;
+using System.Reflection;
+using System.Security.Claims;
+using System.Text;
+
+namespace BusinessLogic
+{
+    public class JwtService : IJwtService
+    {
+        private readonly JwtOptions _JwtOptions;
+
+        public JwtService(IOptions<JwtOptions> jwtOptions)
+        {
+            _JwtOptions = jwtOptions.Value;
+        }
+
+        public string GenerateJwt(Guid userId, string userName)
+        {
+            var claims = new List<Claim>
+            {
+                new Claim(JwtRegisteredClaimNames.Sub, userId.ToString()),
+                new Claim(JwtRegisteredClaimNames.Name, userName)
+            };
+
+            foreach (var role in UserConstants.Roles)
+            {
+                claims.Add(new Claim(ClaimTypes.Role, role));
+            }
```

Comment:
We cannot do this way. If we do so, then all users will have ALL application roles and will have access to all endpoints.
I suggest you passing to this method IEnumerable of roles and add them this way - you only need to add roles of the user you are creating token for